File: database-sample/app/src/main/java/de/cotech/hw/database/sample/MyCustomApplication.java
package de.cotech.hw.database.sample;


import android.app.Application;

import de.cotech.hw.SecurityKeyManager;
import de.cotech.hw.SecurityKeyManagerConfig;
import de.cotech.hw.openpgp.OpenPgpSecurityKeyConnectionMode;
import de.cotech.hw.openpgp.OpenPgpSecurityKeyConnectionModeConfig;

public class MyCustomApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        OpenPgpSecurityKeyConnectionModeConfig pgpConfig = new OpenPgpSecurityKeyConnectionModeConfig.Builder()
                .build();
        OpenPgpSecurityKeyConnectionMode.setDefaultConfig(pgpConfig);

        SecurityKeyManagerConfig config = new SecurityKeyManagerConfig.Builder()
                .setEnableDebugLogging(BuildConfig.DEBUG)
                .build();
        SecurityKeyManager.getInstance().init(this, config);
    }
}

File: pgp-piv-ssh-sample/app/src/main/java/de/cotech/hw/ssh/sample/MyCustomApplication.kt
package de.cotech.hw.ssh.sample

import android.app.Application
import de.cotech.hw.SecurityKeyManager
import de.cotech.hw.SecurityKeyManagerConfig
import org.bouncycastle.jce.provider.BouncyCastleProvider
import java.security.Security

class MyCustomApplication : Application() {

    companion object {
        const val TAG = "SSH"
    }

    override fun onCreate() {
        super.onCreate()

        val securityKeyManager = SecurityKeyManager.getInstance()
        val config = SecurityKeyManagerConfig.Builder()
                .setEnableDebugLogging(BuildConfig.DEBUG)
                .build()
        securityKeyManager.init(this, config)

        // required to make SSHJ work with modern ciphers
        Security.removeProvider("BC") //first remove default os provider
        Security.addProvider(BouncyCastleProvider()) //add new provider
    }
}
